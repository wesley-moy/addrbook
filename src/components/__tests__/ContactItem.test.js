import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ContactItem from '../ContactItem.react.js';

describe('ContactItem Component', () => {
  let contactItem;
  let event;
  let contact = {
    id: 1,
    firstName: 'test',
    lastName: 'testson',
    email: 'test@mail.com',
  };


  const removeContact = jest.fn();

  describe('Render Tests', () => {
    beforeEach(() => {
      contactItem = shallow(<ContactItem
        contact={contact}
        firstName={contact.firstName}
        lastName={contact.lastName}
        email={contact.email}
      />);
      removeContact.mockClear();
    });

    it('Renders correctly', () => {
      expect(shallowToJson(contactItem)).toMatchSnapshot();
    });

  });

  describe('test with undefined handleRemove', () => {
    beforeEach(() => {
      event = {
        preventDefault: () => {},
      };
    });

    it('handles remove without handleRemove prop', () => {
      contactItem.instance().handleRemove(event);
      expect(removeContact).not.toHaveBeenCalled();
    });
  });

  describe('test with defined handleRemove', () => {
    beforeEach(() => {
      contactItem = shallow(<ContactItem
        contact={contact}
        firstName={contact.firstName}
        lastName={contact.lastName}
        email={contact.email}
        handleRemove={removeContact}
      />);
      removeContact.mockClear();
      event = {
          preventDefault: () => {},
      };
    });

    it('handles remove with handleRemove prop', () => {
      contactItem.instance().handleRemove(event);
      expect(removeContact).toHaveBeenCalled();
    });
  });


});
