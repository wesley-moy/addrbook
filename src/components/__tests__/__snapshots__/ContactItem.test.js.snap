// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem Component Render Tests Renders correctly 1`] = `
<div
  className="contact-item-component"
>
  <div
    className="contact-buttons"
  >
    <Button
      label="âœŽ"
      linkTo="updateContact/1"
    />
  </div>
  <div
    className="contact-name"
  >
    <span>
      testson
      , 
    </span>
    <span>
      test
    </span>
  </div>
  <div
    className="contact-email"
  >
    <span>
      test@mail.com
    </span>
  </div>
</div>
`;
